src/DataCollectors/Microsoft.TestPlatform.Extensions.EventLogCollector/Properties/TypesToLoadAttribute.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Microsoft.TestPlatform.Extensions.EventLogCollector;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Utilities;
+
+[assembly: TypesToLoad(typeof(EventLogDataCollector))]
+
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel.Utilities
+{
+    /// <summary>
+    /// Custom Attribute to specify the exact types which should be loaded from assembly
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Assembly, Inherited = false, AllowMultiple = false)]
+    internal sealed class TypesToLoadAttribute : Attribute
+    {
+        public TypesToLoadAttribute(params Type[] types)
+        {
+            this.Types = types;
+        }
+
+        public Type[] Types { get; }
```

Comment:
Why not have the types initialize right here?
`public Type[] Types = {typeof(EventLogDataCollector)}`. In this case no need for constructor and above statement.

Reply:
I didn't want to change the implementation and keep it somewhat consistent in all adapters. This way actual assembly-level attribute is cleaner,